openvanilla/fcitx5-mcbopomofo
# Fix issues raised by cpplint

This PR resolved issues raised by `cpplint`.

I addressed them by categorized in each commit. Here's the list of the change:

- use C++ style `static_cast`
- update `README.md` with linter instructions
- add missing includes
- group `<fstream>` with other standard libs
- fix misuse of <> for gtest
- reorganize filter list with blocklist only and sort with alphabetic order.

A few remarks:

1. These checks are turned off since they have conflicts with clang-format:

`-build/include_alpha,-build/include_order,-build/include_subdir,-whitespace`

2. `readability/nolint` is disabled as `cpplint` doesn't recognize clang-tidy's flag.

```text
src/Engine/UserOverrideModel.cpp:57:  Unknown NOLINT error category: readability-magic-numbers  [readability/nolint] [5]
```

3. disable build/c++11 as chrono is banned by cpplint.

4. disable runtime/references check as it may be controversial. @lukhnos I'm not sure if we'd like to enforce this one?

```text
src/Engine/AssociatedPhrases.h:51:  Is this a non-const reference? If so, make const or use a pointer: std::string_view& k  [runtime/references] [2]
src/Engine/AssociatedPhrases.h:51:  Is this a non-const reference? If so, make const or use a pointer: std::string_view& v  [runtime/references] [2]
```
